Doc/c-api/intro.rst

```diff
@@ -236,8 +236,8 @@ assignments, scope rules, and argument passing), it is only fitting that they
 should be represented by a single C type.  Almost all Python objects live on the
 heap: you never declare an automatic or static variable of type
 :c:type:`PyObject`, only pointer variables of type :c:type:`PyObject*` can  be
-declared.  The sole exception are the type objects; since these must never be
-deallocated, they are typically static :c:type:`PyTypeObject` objects.
+declared.  The sole exception are the type objects; they may be static
+:c:type:`PyTypeObject` objects.
```

Comment:
"The sole exception are the type objects" looks wrong to me. Almost all types in Python are allocated on the heap.

Only **static** types are never deallocated.

I'm not sure why these kinds of technical details are giving in an introduction. It would be better to move them to https://docs.python.org/dev/c-api/refcounting.html no?

Reply:
Or at least, copied there.

Reply:
Hm, I am not find the docs of `PyHeapTypeObject`. Maybe add those description info in  it.

Reply:
I'm not certain what to do with this change. The whole paragraph could be revisited, as Victor hints to. I'm not sure if this change makes things clearer for the reader.

@encukou: thoughts?

Reply:
To be technically correct, it's not just types – `PyModuleDef` is also a `PyObject*` (though I assume that's for historical reasons only). But that's too much detail here.